text/0092-logs-vision.md

```diff
@@ -73,6 +73,15 @@ Unified collection is important for the following reasons:
   of resources where the data comes from or by user-defined attributes),
   enabling correct correlation across Resource dimensions later on the backend.
 
+#### Performance
+
+The unified collector must be performant; its resource utilization should be minimal for most use cases, but must also handle high throughput when needed.
+
+We have identified three sets of use cases for the collector with differing requirements:
+- *Side-car*: There are a set of use cases where a single instance of the collector will be deployed alongside a single instance of an application- a 1:1 relationship. This set includes many serverless environments. A single instance of an application may produce telemetry events (logs, metrics, traces) measured in 10s to 1000s per second. For these use cases, the collector's resource consumption should not exceed 100 MB memory, or 20% of a CPU on a modern server. We will try to optimize the collector so that for many users in this group, the collector will consume less than 50 MB of memory and less than 10% of a CPU.
```

Comment:
Collector CPU and RAM usage normally is load-dependent. Performance requirements that do not specify the load can be of 2 types:
- Resource usage under zero load - idle mode.
- Resource usage under overload, i.e. maximum allowed CPU and RAM usage no matter how high the load (up to certain limit until it saturates the ability of the process to trim the input load).

I am not sure which of these cases the requirement in this PR attempts to cover or if it is about some other case. This needs clarification.

There is another case for performance requirements: the normal operating case, where the load is above zero but below the point where overload occurs. The requirement for this case must include the load description - the rate of incoming data and the composition of the data (e.g. 1000 spans per second, each span with 10 attributes of average (random) length of 20 bytes, etc). Given that the resource consumption may have non-linear correlation with load rate it may be necessary to specify multiple data points for this. We will also likely need different requirements for traces, metrics and logs (although we should try to be consistent across signals where we can). Furthermore, resource usage is protocol dependent, so this needs to say which protocol we want to test. Finally, performance is highly affected by processors, presumably these are numbers without any processing, in passthrough mode. This needs to be called out specifically or we need to call out what processing is part of the benchmarks.

There are other cases that may be interesting from performance perspective. I think these 3 are most important.